### DISCUSSION TITLE
Using nuc9i9qnx instead of nuc8i7hvk - for HaC Mini

### DISCUSSION THREAD

**Discussion by helmehraz** [osy86/HaC-Mini#562] [General]
**As mentioned in the title this is not a bug report but a king of discussion thread**

there are some similarities between the following models [nuc8i7hvk & nuc9i9qnx]! anyone ever tried to implement the work described here on a nuc9i9qnx nuc ? do you think this can be done with some minors tweaks?

Thanks

**Comment by RogerChern** [comment]
I don't see why this is a BUG? Espetially you just mannual labelled it?
If you think some minor tweaks could work, you may have a try yourself.

**Comment by helmehraz** [comment]
> I don't see why this is a BUG? Espetially you just mannual labelled it?
> If you think some minor tweaks could work, you may have a try yourself.

@RogerChern My bad the label was a mistake ! and i would like change it to "discussion" but i do not have the necessary rights to do it !

**Comment by jarryji** [comment]
I managed installed Catalina 10.15.7 on my NUC9i7QNX with minor modification of one guy's EFI made for nuc9vxqnx, thanks to him. even it's not free.

**Comment by helmehraz** [comment]
@jarryji Good to know :) and everything is working even the two front usb ports and hotplug thunderbolt ?

on my side I also managed to install Catalina, it works quite well! I tested a number of things and others have not yet test them. **here is my current status :** 

**Working Hardware:** 
- [x] Ethernet (both ports [I219LM & I210AT])
- [x] HDMI/DP audio
- [x] USB A ports (Black & Blue) [Front & Rear]
- [x] SD card slot
- [x] NVMe/SATA SSD
- [x] Sleep/Resume
- [x] BT (with the built-in card)
- [x] Speaker/TOSLINK jack on the back panel
- [X] HDMI/DP audio
- [x] Headphone on the front panel
- [x] Wifi 6 (using built-in card [AX200]) "But speed is not that good"
- [x] Analog Microphone (front panel jack)
- [x] Thunderbolt 3 (HotPlug) (without flashing the firmware, only ssdts) (tested using a pegasus2 r4 and TBT3toTBT adapter)
- [X] Multiple displays (up to Three 4K displays) "HDMI v2.0 enabled"
- [X] Hardware Acceleration and Decoding
- [X] USB C ports 
- [X] Handoff, Continuity, Universal Clipboard, Apple Watch unlock (with BCM94360CS2 module & M.2 Key B+M Adapter)

**Working Software:** 
 - [x] Installer, App Store, app updates, OS updates
 - [x] iMessage, iCloud, Siri, iTunes, other services
 - [x] Time Machine
 - [x] Parallels/VMWare and virtualBox
 
**Not Working Software:** 
- [ ] AppleTv App,

**Comment by jarryji** [comment]
usb b type a both front and back works fine, but type-c doesnt work (quick charger works), 
Thunderbolt not tested probably not working.
front earphone jack works when earphone pluged in.
microphone doesn't work.
bt works.
uhd 630 3d game Acceleration frame rate only 20 fps while playing CSGO.

btw: for intel ax200 there have third part driver available. I havent test yet.
i have attached my EFI, see if it's can help you.
[EFI.zip](https://github.com/osy86/HaC-Mini/files/5597905/EFI.zip)

**Comment by helmehraz** [comment]
@jarryji  thank u very much dude ! tonight i will have a look on your EFI

**Comment by helmehraz** [comment]
@jarryji I did give a try to your EFI, but i couldn't get it to work ... i couldn't even boot :(

I haven't spent a lot of time setting up my HackMini but just enough to get all the USB ports working using this [guide.](https://dortania.github.io/OpenCore-Post-Install/usb/intel-mapping/intel.html).

I also took a look at Osyx86 articles to understand how to patch thunderbolt firmware ... but I still have few questions about that.

once my HackMini finished I plan to share my EFI with the community :)

**Comment by helmehraz** [comment]
Hi @osy86, I'm sorry to bother you with this question, but I wanted to know if I understood everything correctly after reading this [page](https://github.com/osy86/ThunderboltReset/blob/master/PatchingNVM.md)! indeed, my question turn around thunderbolt firmware patching.

I tried to do the execice using the attached binary files  but unfortunately I find more differences than you when I compare the two binary files ... and I wonder where I could have gone wrong.

Please notice that the Apple firmware v33 was found inside an USBCUpdater coming from 10.13.x version instead of 10.14.x

- [Intel_Hades_AR_HR_4C_C0_rev33_W_TI_20180109_SEC1_sign.bin.zip](https://github.com/osy86/HaC-Mini/files/5632103/Intel_Hades_AR_HR_4C_C0_rev33_W_TI_20180109_SEC1_sign.bin.zip)

- [Mac-BE088AF8C5EB4FA2-C0_5.56.0-C0_33.1.bin.zip](https://github.com/osy86/HaC-Mini/files/5632104/Mac-BE088AF8C5EB4FA2-C0_5.56.0-C0_33.1.bin.zip)

here is the list of diffs :
![Untitled 2](https://user-images.githubusercontent.com/26005780/100937301-1d683d00-34f3-11eb-86eb-9933894e8c75.png)



Thanks a lot for your help

**Comment by osy** [comment]
It's been a while so I don't completely recall. But I think I started with all the changes then one by one I reverted it and tested to see if anything changed (everything still works) then ended up with a minimal set of changes that actually affected functionality.

**Comment by helmehraz** [comment]
Thanks @osy86 👍🏻

**Comment by jarryji** [comment]
I am sure that this EFI works with my nuc9i7qnx and I have confirmed that USB type C works well on my NUC9

**Comment by helmehraz** [comment]
I have no doubts about it ... I'm just trying to understand why your EFI as is,  is not working in my case! and I think (perhaps) I found it by reading your answer twice ;)

your nuc model is i7 processor (nuc9i7qnx) and mine is an i9 (nuc9i7qnx) !

Luck u :) My thunderbolt are not working yet :(

**Comment by helmehraz** [comment]
Hi @jarryji please could you try this EFI and tell me if it solve your microphone, builtin wifi and  CSGO hardware acceleration issues ?

Thanks.

[Intel_Quartz_Canyon_EFI.zip](https://github.com/osy/HaC-Mini/files/5730256/Intel_Quartz_Canyon_EFI.zip)

**Comment by jarryji** [comment]
No, it's not work on my nuc9i7qnx.
thank you anyways.
